primefaces/src/main/resources/META-INF/resources/primefaces/datepicker/0-datepicker.js

```diff
@@ -278,6 +278,11 @@
             return firstDayOfWeek > 0 ? 7 - firstDayOfWeek : 0;
         },
 
+        getSaturdayIndex: function () {
+            var firstDayOfWeek = this.options.locale.firstDay !== undefined ? this.options.locale.firstDay : this.options.locale.firstDayOfWeek;
+            return firstDayOfWeek > 0 ? 7 - firstDayOfWeek - 1 : 6;
```

Comment:
Isn't this just
```suggestion
            return 7 - firstDayOfWeek - 1;
```
?